llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h

```diff
@@ -274,42 +268,20 @@ template <unsigned Opcode, typename... RecipeTys>
 using ZeroOpRecipe_match =
     Recipe_match<std::tuple<>, Opcode, false, RecipeTys...>;
 
-template <typename Op0_t, unsigned Opcode, typename... RecipeTys>
-using UnaryRecipe_match =
-    Recipe_match<std::tuple<Op0_t>, Opcode, false, RecipeTys...>;
-
-template <typename Op0_t, unsigned Opcode>
-using UnaryVPInstruction_match =
-    UnaryRecipe_match<Op0_t, Opcode, VPInstruction>;
-
 template <unsigned Opcode>
 using ZeroOpVPInstruction_match = ZeroOpRecipe_match<Opcode, VPInstruction>;
 
 template <typename Op0_t, unsigned Opcode>
```

Comment:
```suggestion
template <unsigned Opcode, typename Op0_t>
```

to match VPInstruction_match.

Reply:
Should we go through and do the same for AllBinaryRecipe_match too

If we're going to do that I might as well see if I can get rid of those too, and just replace them with AllRecipe_match

Reply:
Yeah, I think having a single AllRecipe_match is the way to go, which can then be specialized if necessary?

Reply:
Done in 7c860258e6c99ec55c74d3c97807dd3478df673b